code.js

```diff
@@ -1,3 +1,12 @@
 function fib(n) {
-    return n;
+    if (n === 0) return [0];
+    if (n === 1) return [0, 1];
+    let fibArray = [0, 1];
+    addFibNumbers(2);
+    function addFibNumbers(i) {
```

Comment:
This makes it really unclear how you're computing the $n$th Fibonacci number.

Reply:
If $n$ is greater then 1 it's creating an array with (0,1) then addFibNumbers(i) is set to 2. So the next Fibonacci number is calculated by adding the array (i-1)+(i-2) and adds that number to the array. It does that until "i" is greater then n. To get the $n$th Fibonacci number I should of changed the return to "return fibArray[fibArray.length -1]" but the code fails when that's added.

Reply:
Can you restructure your code to make it clearer what happens please?